Cesium3DTilesContent/include/Cesium3DTilesContent/LegacyUtilities.h

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include <Cesium3DTilesContent/GltfConverters.h>
+#include <CesiumAsync/IAssetAccessor.h>
+#include <CesiumAsync/IAssetRequest.h>
+#include <CesiumGltf/AccessorView.h>
+#include <CesiumGltf/PropertyTransformations.h>
+#include <CesiumUtility/ErrorList.h>
+
+#include <glm/fwd.hpp>
+#include <rapidjson/document.h>
+
+#include <cstdint>
+#include <optional>
+#include <vector>
+
+namespace CesiumGltf {
+class Model;
+class Buffer;
+} // namespace CesiumGltf
+
+namespace Cesium3DTilesContent {
+
+struct ByteResult {
+  std::vector<std::byte> bytes;
+  CesiumUtility::ErrorList errorList;
+};
+
+CesiumAsync::Future<ByteResult>
+get(const ConverterSubprocessor& subprocessor, const std::string& relativeUrl);
+
+namespace LegacyUtilities {
+std::optional<uint32_t> parseOffset(
+    const rapidjson::Document& document,
+    const char* semantic,
+    CesiumUtility::ErrorList& errorList);
+
+typedef bool (rapidjson::Value::*ValuePredicate)() const;
+
+template <typename T> bool isValue(const rapidjson::Value& value);
+template <typename T> T getValue(const rapidjson::Value& value);
+
+template <typename T> std::optional<T> getOpt(const rapidjson::Value& value) {
+  if (isValue<T>(value)) {
+    return std::make_optional(getValue<T>(value));
+  }
+  return {};
+}
+
+template <typename T>
+std::optional<T>
+getValue(const rapidjson::Document& document, const char* semantic) {
+  const auto valueIt = document.FindMember(semantic);
+  if (valueIt == document.MemberEnd() || !isValue<T>(valueIt->value)) {
+    return {};
+  }
+  return std::make_optional(getValue<T>(valueIt->value));
+}
+
+template <> inline bool isValue<bool>(const rapidjson::Value& value) {
+  return value.IsBool();
+}
+
+template <> inline bool getValue<bool>(const rapidjson::Value& value) {
+  return value.GetBool();
+}
+
+template <> inline bool isValue<uint32_t>(const rapidjson::Value& value) {
+  return value.IsUint();
+}
+
+template <> inline uint32_t getValue<uint32_t>(const rapidjson::Value& value) {
+  return value.GetUint();
+}
+
+bool validateJsonArrayValues(
+    const rapidjson::Value& arrayValue,
+    uint32_t expectedLength,
+    ValuePredicate predicate);
+
+std::optional<glm::dvec3>
+parseArrayValueDVec3(const rapidjson::Value& arrayValue);
+
+std::optional<glm::dvec3>
+parseArrayValueDVec3(const rapidjson::Document& document, const char* name);
+
+int32_t createBufferInGltf(CesiumGltf::Model& gltf);
+int32_t
+createBufferInGltf(CesiumGltf::Model& gltf, std::vector<std::byte>&& buffer);
```

Comment:
These are named the same function but do slightly different things; can they be named differently for clarity?
```suggestion
int32_t createEmptyBufferInGltf(CesiumGltf::Model& gltf);

int32_t
createBufferInGltf(CesiumGltf::Model& gltf, std::vector<std::byte>&& buffer);
```

Reply:
I would have said that it is one function with an optional argument, but it has to be written as two because the optional argument is an rvalue. Actually, I think I see what can be done.

Apropos of nothing, this is one of the many functions I copied from PntsToGltfConverter.cpp. Do you think it would be worthwhile in this PR to refactor PntsToGltfConverter to use the common versions?

Reply:
Yeah, if you're willing to refactor the existing converters, go ahead! :pray: